javascript/index.js

```diff
@@ -1,13 +1,40 @@
-import createSubscription from './modules/createSubscription';
 import reframeTime from './modules/reframeTime';
 import scrollConversationToBottom from './modules/scrollConversationToBottom';
-import setScrollListener from './modules/setScrollListener';
-import setSendMessageListener from './modules/setSendMessageListener';
+
+const conversationHistory = document.querySelector('.active-admin-chat__conversation-history');
+const sendMessageForm = document.querySelector('.active-admin-chat__send-message-form');
+let firstElement = null;
+let loadingMessages = false;
+let scrolling = false;
 
 document.addEventListener('DOMContentLoaded', () => {
   reframeTime();
   scrollConversationToBottom();
-  const channelSubscription = createSubscription();
-  setSendMessageListener(channelSubscription);
-  setScrollListener();
 });
+
+document.addEventListener('motion:render', () => {
+  reframeTime();
+
+  if (loadingMessages) {
+    loadingMessages = false;
+    firstElement.scrollIntoView();
+  } else if (!scrolling) {
+    scrollConversationToBottom();
+  }
+});
+
+conversationHistory.addEventListener('scroll', event => {
```

Comment:
needed to hack the `scroll` event listener to NOT trigger motion unless I reached the top of the conversation. This is a bit unfortunate, I'd have like to have a way to configure the handler more precisely.

Reply:
In the case, I would remove the  `data-motion="scroll->"` handler and trigger motion programatically.

```javascript
// The Motion client instance is the default export of the JavaScript initializer 
import motion from "./motion";

// This is how you trigger a Motion
motion.getComponent(element).performMotion(motionName, event)
```

You could also bind Motion to a custom event and fire that here instead.

Reply:
nice! I'll try that out